Tools/TestWebKitAPI/Tests/WebKitCocoa/ObscuredContentInsets.mm

```diff
@@ -210,8 +210,38 @@ - (void)userContentController:(WKUserContentController *)userContentController d
     EXPECT_EQ(WebCore::serializationForCSS(colorAfterChangingBackground), "rgb(34, 34, 34)"_s);
 }
 
+TEST(ObscuredContentInsets, TopOverhangColorExtensionLayer)
+{
+    RetainPtr webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 600, 400)]);
+
+    [webView _setAutomaticallyAdjustsContentInsets:NO];
+    [webView _setObscuredContentInsets:NSEdgeInsetsMake(100, 0, 0, 0) immediate:NO];
+    [webView waitForNextPresentationUpdate];
+
+    [webView synchronouslyLoadTestPageNamed:@"top-fixed-element"];
+    [webView waitForNextPresentationUpdate];
+
+    __block RetainPtr<CALayer> colorExtensionLayer;
+    [webView forEachCALayer:^(CALayer *layer) {
+        if ([layer.name containsString:@"top overhang"])
+            colorExtensionLayer = layer;
+    }];
+
+    auto expectedComponents = CGColorGetComponents([webView _sampledTopFixedPositionContentColor].CGColor);
+    auto actualComponents = CGColorGetComponents([colorExtensionLayer backgroundColor]);
+
+    auto colorExtensionLayerFrame = [colorExtensionLayer frame];
+    EXPECT_FALSE(NSIsEmptyRect(colorExtensionLayerFrame));
+    EXPECT_TRUE(WTF::areEssentiallyEqual<float>(NSWidth(colorExtensionLayerFrame), 600.));
+    EXPECT_TRUE(WTF::areEssentiallyEqual<float>(NSMaxY(colorExtensionLayerFrame), 100.));
+    EXPECT_TRUE(WTF::areEssentiallyEqual<float>(expectedComponents[0], actualComponents[0]));
+    EXPECT_TRUE(WTF::areEssentiallyEqual<float>(expectedComponents[1], actualComponents[1]));
+    EXPECT_TRUE(WTF::areEssentiallyEqual<float>(expectedComponents[2], actualComponents[2]));
+    EXPECT_TRUE(WTF::areEssentiallyEqual<float>(expectedComponents[3], actualComponents[3]));
```

Comment:
We have `TestWebKitAPI::Util::compareColors(CocoaColor *, CocoaColor *, float tolerance = 0.01)` so you can just get the Cocoa representation and try this?

```cpp
EXPECT_TRUE(compareColors([webView _sampledTopFixedPositionContentColor], [colorExtensionLayer backgroundColor]));
```

Reply:
Good call! Will change to use this.